crates/aptos-runtimes/src/thread_manager.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{spawn_named_runtime_with_start_hook, spawn_rayon_thread_pool_with_start_hook};
+#[cfg(target_os = "linux")]
+use libc::{cpu_set_t, sched_setaffinity, CPU_SET};
+use once_cell::sync::{Lazy, OnceCell};
+use rayon::ThreadPool;
+use tokio::runtime::{Handle, Runtime};
+
+pub static THREAD_MANAGER: Lazy<ThreadManager> =
+    Lazy::new(|| ThreadManager::new(ThreadManager::get_thread_config_strategy()));
+
+static THREAD_CONFIG_STRATEGY: OnceCell<ThreadConfigStrategy> = OnceCell::new();
+
+struct ThreadsConfig {
+    num_threads: usize,
+    on_thread_start: Box<dyn Fn() + Send + Sync + 'static>,
```

Comment:
thread is able to take different functions, should we include the function in the thread_config? for example, the same IO pool was used for JMT and adding chunk data.

Reply:
This is the function you want to core when the thread is starting. Usually pinning to cpu cores, or setting thread priorities, etc. If these are different they probably should use different thread pool.